test/e2e/mattermost_test.go

```diff
@@ -122,6 +122,17 @@ func mattermostScaleTest(t *testing.T, f *framework.Framework, ctx *framework.Te
 	err = e2eutil.WaitForDeployment(t, f.KubeClient, namespace, "test-mm", 2, retryInterval, timeout)
 	require.NoError(t, err)
 
+	err = waitForReconcilicationComplete(t, f.Client.Client, namespace, "test-mm", retryInterval, timeout)
+	require.NoError(t, err)
+
+	// scale down again
+	err = f.Client.Get(context.TODO(), types.NamespacedName{Name: "test-mm", Namespace: namespace}, exampleMattermost)
+	require.NoError(t, err)
+
+	exampleMattermost.Spec.Replicas = 1
+	err = f.Client.Update(context.TODO(), exampleMattermost)
+	require.NoError(t, err)
+
```

Comment:
We should wait for the deployment to reach 1 again here, right?

Reply:
true, bad merge thing :(